packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx

```diff
@@ -145,6 +149,20 @@ export class ContextualMenu extends BaseComponent<IContextualMenuProps, IContext
     if (this.props.onMenuOpened) {
       this.props.onMenuOpened(this.props);
     }
+
+    // Due to restraints caused by React 16 where we no can no longer
+    // call the ref values of children component with their own render
+    // function before componentDidMount we have to use this setTimeout
+    // hack to have access to ref. The correct approacj to solve this
+    // problem is to create a comonent for the split button and then
+    // have it add it's own event listener. This is logged on Issue 4522
+    this._async.setTimeout(() => {
+      if (this._splitButtonContainers) {
+        this._splitButtonContainers.forEach((value: HTMLDivElement) => {
+          this._events.on(value, 'pointerdown', this._onPointerDown, true);
```

Comment:
hmm. In code earlier in this PR, you went to great lengtch to validate that onPointerDown was "in" the value, and that the value was not falsey. Why are you not doing that here?

Also this feels REALLY gross. I think the lesson learned here is that the button should have been part of the item, not part of the menu...

What if the menu items change? Is this going to execute again? Are you going to miss registering new items? E.g. the items are lazy loaded.